framework/include/problems/FEProblemBase.h

```diff
@@ -849,6 +849,12 @@ class FEProblemBase : public SubProblem, public Restartable
    */
   ReporterData & getReporterData(ReporterData::WriteKey /*key*/) { return _reporter_data; }
 
+  template <typename T>
+  const T & getReporterValue(const ReporterName & reporter_name,
+                             const std::string & object_name,
+                             const ReporterMode & mode,
+                             const std::size_t time_index = 0) const;
```

Comment:
Can we add some doco here, at least pointing to the ReporterData::getReporterData method. It took me a bit to remember that the "object_name" is the object asking and is simply for error reporting.